### PR TITLE
Fixes #9453 - improve tooltips on puppetclass selection forms

### PR DESCRIPTION
**Pull request by tbrisker**
…s in host/hostgroup/configgroup edit

### CODE REVIEW

File: app/assets/javascripts/class_edit.js [hunk 1]
```diff
@@ -19,12 +19,12 @@ function add_puppet_class(item){
...
   content.attr('id', 'selected_puppetclass_'+ id);
   content.append("<input id='" + type +"_puppetclass_ids_' name='" + type +"[puppetclass_ids][]' type='hidden' value=" +id+ ">");
-  content.children('span').tooltip();
 
   var link = content.children('a');
   var links = content.find('a');
+  var fullname = _.last(link.data('original-title').split(' '))
   links.attr('onclick', 'remove_puppet_class(this)');
-  links.attr('data-original-title', __('Click to undo adding this class'));
+  links.attr('data-original-title', __('Click to remove ')+_.escape(fullname));
```
Comment (domcleal) [inline app/assets/javascripts/class_edit.js]: Use sprintf, not concatenation.

File: app/assets/javascripts/class_edit.js [hunk 2]
```diff
@@ -19,12 +19,12 @@ function add_puppet_class(item){
   var content = $(item).closest('li').clone();
   content.attr('id', 'selected_puppetclass_'+ id);
   content.append("<input id='" + type +"_puppetclass_ids_' name='" + type +"[puppetclass_ids][]' type='hidden' value=" +id+ ">");
-  content.children('span').tooltip();
 
   var link = content.children('a');
   var links = content.find('a');
+  var fullname = _.last(link.data('original-title').split(' '))
```
Comment (domcleal) [inline app/assets/javascripts/class_edit.js]: What string is this splitting precisely?

**Comment by domcleal** [comment]
Please use commas between ticket numbers in the commit message.

File: app/assets/javascripts/class_edit.js (hunk 2 above)
Comment (tbrisker) [inline app/assets/javascripts/class_edit.js]: "Click to add %s", which could be localized so replacing add with remove won't work.

Comment (domcleal) [inline app/assets/javascripts/class_edit.js]: Well neither will splitting it and assuming the class name is after the last space...

**Comment by tbrisker** [comment]
@domcleal addressing your comments led to a bit of refactoring to DRY and remove dead code. Please review, [test] failure is random katello failure.

File: app/helpers/application_helper.rb [hunk 3]
```diff
@@ -64,40 +64,28 @@ def link_to_add_fields(name, f, association, partial = nil, options = {})
...
-    options = klass.name.size > 28 ? {:'data-original-title'=>klass.name, :rel=>'twipsy'} : {}
-    functions_options = { :klass => klass, :host => host, :css_class => ''}
-    text = remove_link_to_function(truncate(klass.name, :length => 28), functions_options)
-    content_tag(:span, text, options).html_safe +
-        remove_link_to_function('', functions_options.merge(:css_class => 'glyphicon glyphicon-minus-sign'))
+  def link_to_remove_puppetclass(klass, type)
+    options = options_for_puppetclass_selection(klass, type)
+    text = remove_link_to_function(truncate(klass.name, :length => 28), options)
+    content_tag(:span, text).html_safe +
+        remove_link_to_function('', options.merge!(:class => 'glyphicon glyphicon-minus-sign'))
```
Comment (domcleal) [inline app/helpers/application_helper.rb]: Use `merge` and not ! in all of these calls, as the result is being passed into the method.

**Comment by domcleal** [comment]
re ticket #2879, it looks like this is already working correctly in develop.  I think there was a change to the CSS of tooltips somewhere else lately, might that have done it?  I can't see what in your PR changes this.

screenshot from 2015-08-24 12-34-00
(develop)

Ditto with #5733, isn't that the same issue?

This most directly seems to be #9453, though you've implemented something different.  I do prefer leaving the class name in the tooltip though, as I think it's more helpful when it's truncated.

**Comment by tbrisker** [comment]
@domcleal It seems like #2879 and #5733 are indeed already fixed in develop, although there are some issues with the tooltips as they are currently. I seem to recall a certain case where they were broken but can't seem to reproduce now. As far as I'm concerned they can both be marked as resolved and I'll edit the commit message for this to only #9453 (or also add another new issue for the other changes in this if you wish).

**Comment by domcleal** [comment]
Yep, please just use #9453, you've improved the wording.  If you can mark the other two with release 1.10.0 and resolved, then that's good.  Thanks.

**Comment by tbrisker** [comment]
@domcleal updated commit message, addressed comments and marked other bugs as resolved.

**Comment by dLobatog** [comment]
@tbrisker I think on the original bug they want it to say "Add this class" and "Remove this class", as it's a bit more 'friendly'. I'd say just change that & I'll merge. Works fine.

**Comment by domcleal** [comment]
I think it's an improvement as it is, just ignore that suggestion?

**Comment by domcleal** [comment]
One reason it's important to keep the class name in the tooltip is because the names in the actual list may be truncated, as there's limited space.  The tooltips allow users to see the long names, which is what the other tickets in the title were originally about (but are currently fixed in develop).

**Comment by dLobatog** [comment]
Merged as acc07525abad7a78b243716dcd6e74feaac95897, thanks @tbrisker!

**Pull request closed without merging**